core/app/models/spree/stock_transfer.rb

```diff
@@ -100,6 +104,18 @@ def close(closed_by)
 
     private
 
+    def ensure_transfer_items_valid
```

Comment:
`AR::Base#valid?` modifies `errors` as a side effect, so it does align with that. I find `ensure` a problematic word in our codebase. Some methods like `ensure_updated_shipments` have major side effects (deleting all shipments), others like `ensure_shipping_address` just perform validations.

Maybe we would all be happy with `validate_transfer_items`?